go/vt/vtgate/safe_session.go

```diff
@@ -16,7 +16,7 @@ import (
 // (the use pattern is 'Find', if not found, then 'Append',
 // for a single shard)
 type SafeSession struct {
-	mu sync.Mutex
+	mu sync.RWMutex
```

Comment:
I'd rather not fix the ones we're not 100% sure contribute. Otherwise we're gonna bloat the code for no reason... I'd vote to revert this one.